lib/ssh_lib.py

```diff
@@ -26,3 +27,23 @@ def generate_instances_ssh_config(ssh_key_path, ssh_config_file, instances):
                  LogLevel='ERROR')
 
     conf.write(ssh_config_file)
+
+
+def wait_for_host_ssh_up(host, timeout_seconds):
+    """
+    Check if a given host is ready for SSH connection within a given number of seconds
+    :param host: Host public DNS or IP address. It can be a combination of "<user>@<host>"
+    :param timeout_seconds: The maximum number of seconds to check for SSH availability.
+                            If timeout is reached, an exception will be thrown.
+    :return: None
+    """
+    count = 0
+    while count < timeout_seconds:
+        if (os.system(f'ssh-keyscan "{host}" > /dev/null 2>&1') >> 8) == 0:
+            print(f'{host} SSH is up! ({count} seconds)')
+            break
+        else:
+            time.sleep(1)
```

Comment:
I might be being a bit picky here, we can approve as it is. But waiting this way we are going to wait more that `timeout_seconds`, as ssh-keyscan takes same time, even if it's small.